From 5256d903fdcb76282de05f67b606d8bae30c13c8 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Fri, 4 Jan 2013 14:41:06 +0100
Subject: [PATCH 09/11] Tethering: Fix bad inclusion of header for
 SoftApController specific to TI solutions

BZ: 73805

As the SoftApController header specific to TI solutions header
was not included, SoftApController class was not properly allocated
and caused memory corruptions.

Change-Id: Idd30525c292730cfa2b054bc50027071a0dfbbe9
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
Reviewed-on: http://android.intel.com:8080/84969
Reviewed-by: Garcia, JeremieX <jeremiex.garcia@intel.com>
Reviewed-by: Khenissi, MaherX <maherx.khenissi@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Android.mk        |    2 +-
 CommandListener.h |    4 ++++
 2 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/Android.mk b/Android.mk
index 5130ab4..966f54e 100644
--- a/Android.mk
+++ b/Android.mk
@@ -31,7 +31,7 @@ LOCAL_SRC_FILES:=                                      \
                   logwrapper.c                         \
                   main.cpp                             \
 LOCAL_SRC_FILES += SoftapControllerTI.cpp
-
+LOCAL_CFLAGS += -DSOFTAPTI
 ifeq ($(INTEL_WIDI), true)
 LOCAL_SRC_FILES += WidiConnectivity.cpp
 LOCAL_CFLAGS += -DINTEL_WIDI
diff --git a/CommandListener.h b/CommandListener.h
index 951a022..4a840f5 100644
--- a/CommandListener.h
+++ b/CommandListener.h
@@ -23,7 +23,11 @@
 #include "TetherController.h"
 #include "NatController.h"
 #include "PppController.h"
+#ifdef SOFTAPTI
+#include "SoftapControllerTI.h"
+#else
 #include "SoftapController.h"
+#endif
 #include "BandwidthController.h"
 #include "IdletimerController.h"
 #include "InterfaceController.h"
-- 
1.7.9

